array.c

```diff
@@ -7888,77 +7888,124 @@ rb_ary_deconstruct(VALUE ary)
  *
- *  == Creating Arrays
- *
- *  You can create an \Array object explicitly with:
- *
- *  - An {array literal}[rdoc-ref:syntax/literals.rdoc@Array+Literals].
- *
- *  You can convert certain objects to Arrays with:
- *
- *  - \Method #Array.
+ *  Although the effective index into an array is always an integer,
+ *  some methods (both within and outside of class \Array)
+ *  accept one or more non-integer arguments that are
+ *  {integer-convertible objects}[rdoc-ref:implicit_conversion.rdoc@Integer-Convertible+Objects].
  *
- *  An \Array can contain different types of objects.  For
- *  example, the array below contains an Integer, a String and a Float:
  *
- *     ary = [1, "two", 3.0] #=> [1, "two", 3.0]
- *
- *  An array can also be created by calling Array.new with zero, one
- *  (the initial size of the \Array) or two arguments (the initial size and a
- *  default object).
- *
- *     ary = Array.new    #=> []
- *     Array.new(3)       #=> [nil, nil, nil]
- *     Array.new(3, true) #=> [true, true, true]
- *
- *  Note that the second argument populates the array with references to the
- *  same object.  Therefore, it is only recommended in cases when you need to
- *  instantiate arrays with natively immutable objects such as Symbols,
- *  numbers, true or false.
- *
- *  To create an array with separate objects a block can be passed instead.
- *  This method is safe to use with mutable objects such as hashes, strings or
- *  other arrays:
- *
- *     Array.new(4) {Hash.new}    #=> [{}, {}, {}, {}]
- *     Array.new(4) {|i| i.to_s } #=> ["0", "1", "2", "3"]
- *
- *  This is also a quick way to build up multi-dimensional arrays:
- *
- *     empty_table = Array.new(3) {Array.new(3)}
- *     #=> [[nil, nil, nil], [nil, nil, nil], [nil, nil, nil]]
+ *  == Creating Arrays
  *
- *  An array can also be created by using the Array() method, provided by
- *  Kernel, which tries to call #to_ary, then #to_a on its argument.
+ *  You can create an \Array object explicitly with:
  *
- *	Array({:a => "a", :b => "b"}) #=> [[:a, "a"], [:b, "b"]]
+ *  - An {array literal}[rdoc-ref:literals.rdoc@Array+Literals]:
+ *
+ *     [1, 'one', :one, [2, 'two', :two]]
+ *
+ *
+ *  - A {%w or %W: string-array Literal}[rdoc-ref:literals.rdoc@25w+and+-25W-3A+String-Array+Literals]:
+ *
+ *      %w[foo bar baz] # => ["foo", "bar", "baz"]
+ *      %w[1 % *]       # => ["1", "%", "*"]
+ *
+ *  - A {%i pr %I: symbol-array Literal}[rdoc-ref:literals.rdoc@25i+and+-25I-3A+Symbol-Array+Literals]:
+ *
+ *      %i[foo bar baz] # => [:foo, :bar, :baz]
+ *      %i[1 % *]       # => [:"1", :%, :*]
+ *
+ *  - \Method Kernel#Array:
+ *
+ *      Array(["a", "b"])             # => ["a", "b"]
+ *      Array(1..5)                   # => [1, 2, 3, 4, 5]
+ *      Array(key: :value)            # => [[:key, :value]]
+ *      Array(nil)                    # => []
+ *      Array(1)                      # => [1]
+ *	Array({:a => "a", :b => "b"}) # => [[:a, "a"], [:b, "b"]]
```

Comment:
Indentation looks off for this example.

Reply:
This looked right in my clone, so I edited it here;  still looks wrong (b/c of Japanese character?).  If you can actually fix it here (instead of suggested change), I'll be grateful.